### ISSUE TITLE
Add support for JWS Detached 

### ISSUE THREAD

**Issue by lrosenthol** [Thalhammer/jwt-cpp#55]
I have a use case where it I need to produce a JWT where the payload data shouldn't be included in the token itself, as described [here in the spec](https://tools.ietf.org/html/rfc7515#appendix-F).

**Comment by Thalhammer** [comment]
> Note that this method needs no support from JWS libraries, as applications can use this method by modifying the inputs and outputs of standard JWS libraries.

If you want to do this you simply need to split the resulting token on the dot and remove the middle part. Before verification you insert it back.

**Comment by lrosenthol** [comment]
True, but I ended up doing it in a more efficient (at least to me) way.  Changes are in my fork at https://github.com/lrosenthol/jws-jwt-cpp
